lib/web_ui/lib/src/engine/platform_views/content_manager.dart

```diff
@@ -0,0 +1,185 @@
+/// View by its `viewType`.
+/// * `contents`: The result [html.Element] of calling a `factory` function.
+/// * `slots`: A special HTML tag that will be used by the framework to "reveal"
+/// the `contents` of a Platform View.
+///
+/// This class keeps a registry of `factories`, `slots` and `contents` so the
+/// framework can CRUD Platform Views as needed, regardless of the rendering backend.
+class PlatformViewManager {
+  // The factory functions, indexed by the viewType
+  final Map<String, Function> _factories = {};
+
+  // The references to <slot> tags, indexed by their framework-given ID.
+  final Map<int, html.Element> _slots = {};
+
+  // The references to content tags, indexed by their framework-given ID.
+  final Map<int, html.Element> _contents = {};
+
+  String _getSlotName(int viewId) {
+    return 'flt-pv-slot-$viewId';
+  }
+
+  /// Returns `true` if the passed in `viewType` has been registered before.
+  ///
+  /// See [registerViewFactory] to understand how factories are registered.
+  bool knowsViewType(String viewType) {
+    return _factories.containsKey(viewType);
+  }
+
+  /// Returns `true` if the passed in `viewId` has been rendered (and not disposed) before.
+  ///
+  /// See [renderContent] and [renderSlot] to understand how platform views are
+  /// rendered.
+  bool knowsViewId(int viewId) {
+    return _contents.containsKey(viewId);
+  }
+
+  /// Registers a `factoryFunction` that knows how to render a Platform View of `viewType`.
+  ///
+  /// `viewType` is selected by the programmer, but it can't be overridden once
+  /// it's been set.
+  ///
+  /// `factoryFunction` needs to be a [PlatformViewFactory].
+  bool registerFactory(String viewType, Function factoryFunction) {
+    assert(factoryFunction is PlatformViewFactory ||
+        factoryFunction is ParameterizedPlatformViewFactory);
+
+    if (_factories.containsKey(viewType)) {
+      return false;
+    }
+    _factories[viewType] = factoryFunction;
+    return true;
+  }
+
+  /// Creates the HTML markup for the `slot` of a Platform View.
+  ///
+  /// The result of this call is cached in the `_slots` Map, so it can be accessed
+  /// later for CRUD operations.
+  ///
+  /// The resulting DOM for a `slot` looks like this:
+  ///
+  /// ```html
+  /// <flt-platform-view-slot style="...">
+  ///   <slot name="..." />
+  /// </flt-platform-view-slot>
+  /// ```
+  ///
+  /// The inner `SLOT` tag is standard HTML to reveal an element that is rendered
+  /// elsewhere in the DOM. Its `name` attribute must match the value of the `slot`
+  /// attribute of the contents being revealed (see [renderContent].)
+  ///
+  /// The outer `flt-platform-view-slot` tag is a simple wrapper that the framework
+  /// can position/style as needed.
+  ///
+  /// (When the framework accesses a `slot`, it's really accessing its wrapper
+  /// `flt-platform-view-slot` tag)
+  html.Element renderSlot(int viewId) {
```

Comment:
As discussed offline, let's make this a top-level function, or inline into the renderers.

Reply:
Done as a top-level fn.